helix-core/src/main/java/org/apache/helix/monitoring/mbeans/dynamicMBeans/DynamicMBeanProvider.java

```diff
@@ -59,6 +55,7 @@
 
   // Attribute name to the DynamicMetric object mapping
   private final Map<String, DynamicMetric> _attributeMap = new HashMap<>();
+  private final Object _lock = new Object();
```

Comment:
Have you tried what we discussed that we don't lock at all? Since the strong consistency is not possible anyway.

Reply:
Discussed offline. Will consider making it simpler and removing the lock.